bundles/admin/admin-hierarchical-layerlist/components/Group.js

```diff
@@ -0,0 +1,266 @@
+    /* PRIVATE METHODS
+    *******************************************************************************************************************************/
+
+    /*******************************************************************************************************************************
+    /* PUBLIC METHODS
+    *******************************************************************************************************************************/
+    /**
+     * Gets popup configuration obhject
+     * @method getGroupAddingPopupConf
+     * @param  {Object}                tool     jQuery object
+     * @param  {String}                id       group id, if defined then also show delete button
+     * @param  {Integer}               parentId parent group id
+     * @param  {Object}                opts     options:
+     *                                          {
+     *                                              selectable: true,
+     *                                              locale: {
+     *                                                  fi: '',
+     *                                                  en: '',
+     *                                                  sv: ''
+     *                                              }
+     *                                          }
+     * @return {Object}                         popup configuration:
+     *                                          {
+     *                                              popup: null,
+     *                                              buttons: null,
+     *                                              messages: null
+     *                                          }
+     */
+    getGroupAddingPopupConf: function(tool, id, parentId, opts) {
+        var me = this;
+        var returnObject = {
+            popup: null,
+            buttons: [],
+            message: null
+        };
+        var popup = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        var errorDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        errorDialog.setId('admin-hierarchical-layerlist-group-error-popup');
+        returnObject.popup = popup;
+
+        // check at if editing group (id defined)
+        // then also add delete button
+        if (id) {
+            var btnDelete = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            btnDelete.setTitle(me.locale('buttons.delete'));
+            btnDelete.addClass('delete');
+            btnDelete.setHandler(function() {
+                // check at group has no layers or subgroups
+                var group = me.layerService.getAllLayerGroups(id);
+                var isLayersOrGroups = (group && group.layers && group.layers.length > 0) ? true : false;
+                if (group.groups && group.groups.length > 0) {
+                    isLayersOrGroups = true;
```

Comment:
This reads like Q: "Are these layers or groups?" A: "Yes" :) If I understand this correctly you want to check if the group has layers or subgroups? These could be functions in the group object: hasLayers() and hasSubgroups(). It would make this more readable:

    if (group.hasLayers() || group.hasSublayers()) { showError(); }